core/templates/domain/exploration/editable-exploration-backend-api.service.ts

```diff
@@ -16,152 +16,177 @@
  * @fileoverview Service to send changes to a exploration to the backend.
  */
 
-require('domain/exploration/read-only-exploration-backend-api.service.ts');
-require('domain/utilities/url-interpolation.service.ts');
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-require(
-  'pages/exploration-player-page/exploration-player-page.constants.ajs.ts');
+import { HttpClient } from '@angular/common/http';
+import { ReadOnlyExplorationBackendApiService } from
+  'domain/exploration/read-only-exploration-backend-api.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { ExplorationBackendDict } from 'domain/exploration/ExplorationObjectFactory';
+import { DraftExplorationResponse } from 'pages/exploration-editor-page/services/exploration-data.service';
+import { AppConstants } from 'app.constants';
+import cloneDeep from 'lodash/cloneDeep';
 
-angular.module('oppia').factory('EditableExplorationBackendApiService', [
-  '$http', '$q', 'ReadOnlyExplorationBackendApiService',
-  'UrlInterpolationService', 'EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE',
-  'EDITABLE_EXPLORATION_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, ReadOnlyExplorationBackendApiService,
-      UrlInterpolationService, EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE,
-      EDITABLE_EXPLORATION_DATA_URL_TEMPLATE) {
-    var _fetchExploration = function(
-        explorationId, applyDraft, successCallback, errorCallback) {
-      var editableExplorationDataUrl = _getExplorationUrl(
-        explorationId, applyDraft);
-      $http.get(editableExplorationDataUrl).then(function(response) {
-        var exploration = angular.copy(response.data);
-        if (successCallback) {
-          successCallback(exploration);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableExplorationBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private readOnlyExplorationBackendApiService:
+    ReadOnlyExplorationBackendApiService,
+    private urlInterpolationService: UrlInterpolationService) {}
+
+  private _getExplorationUrl(
+      explorationId: string,
+      applyDraft: boolean): string {
+    if (applyDraft) {
+      return this.urlInterpolationService.interpolateUrl(
+        AppConstants.EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE, {
+          exploration_id: explorationId,
+          apply_draft: JSON.stringify(applyDraft)
         }
+      );
+    }
+    return this.urlInterpolationService.interpolateUrl(
+      AppConstants.EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+        exploration_id: explorationId
+      }
+    );
+  }
+
+  private _fetchExploration(
+      explorationId: string,
+      applyDraft: boolean,
+      successCallback:
+      (value: ExplorationBackendDict | DraftExplorationResponse) => void,
```

Comment:
How did you determine these types? Looking at the backend handler `editor.ExplorationHandler` it seems like the return type is fixed (and different from what is mentioned here). Please check.

Ditto in L66.